chore(sentry): enable performance monitoring

gauthierpetetin opened the pull request:
## **Description**

Setting a value for `tracesSampleRate` enables performance monitoring in Sentry. Once performance monitoring is enabled, transactions are sent to Sentry every time a user loads a page or navigates within the app. Since the amount of traffic the app gets is important, this means a lot of transactions are sent. By setting `tracesSampleRate` to a value lower than 1.0, we reduce the volume of transactions to a more reasonable amount.

[Sentry documentation](https://docs.sentry.io/platforms/javascript/performance/#configure)
[Mobile implementation](https://github.com/MetaMask/metamask-mobile/blob/756209bbbd4a79955926d19dcd4a646b813275b2/app/util/sentryUtils.js#L156C5-L156C30)

Enabling Sentry performance monitoring will allow us to add `app load time` to our [North Star metrics dashboards](https://lookerstudio.google.com/u/1/reporting/e7e8f90e-72aa-4128-ae01-6305bf3393f4/page/p_du94w6t4bd), like we do for Mobile.
<img width="523" alt="Screenshot 2023-11-29 at 17 12 18" src="https://github.com/MetaMask/metamask-extension/assets/7071180/dfa71fef-ffb1-4cb2-8297-96aa33faf532">

## **Related issues**

- None

## **Manual testing steps**

1. Create your own Sentry instance
2. Go to Sentry Performance dashboard and you shall see no transaction
3. Select `Settings` in the sidebar menu, then select `Projects` in the secondary menu. Click your project then select `Client Keys (DSN)` from the secondary menu. Click the `Configure` button on the `Client Keys` page and copy your `Project Id` and `Public Key`.
4. Add/replace `SENTRY_DSN_DEV` variable in `.metamaskrc`
    ```
    SENTRY_DSN_DEV=https://{SENTRY_PUBLIC_KEY}@sentry.io/{SENTRY_PROJECT_ID}
    ```

5. Build the project to the `./dist/` folder with `yarn dist`
6. Add Extension to your browser
7. Open Extension and do some actions
8. Kill the Extension
9. Go to Sentry Performance dashboard and you shall see a few transactions

## **Screenshots/Recordings**

### **Before**

<img width="1680" alt="Screenshot 2023-11-29 at 16 31 20" src="https://github.com/MetaMask/metamask-extension/assets/7071180/d40b6952-03cf-4158-948d-69e5e8011653">

### **After**

<img width="1680" alt="Screenshot 2023-11-29 at 17 01 04" src="https://github.com/MetaMask/metamask-extension/assets/7071180/0781fe7b-006d-4e89-a53b-581dfc16f212">

<img width="1679" alt="Screenshot 2023-11-29 at 17 00 26" src="https://github.com/MetaMask/metamask-extension/assets/7071180/2723590a-59e8-438f-b451-bd30751deffd">

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

gauthierpetetin committed:
feat(sentry): enable performance monitoring

Setting a value for tracesSampleRate enables performance monitoring in Sentry. Once performance monitoring is enabled, transactions are sent to Sentry every time a user loads a page or navigates within the app. Since the amount of traffic the app gets is important, this means a lot of transactions are sent. By setting tracesSampleRate to a value lower than 1.0, we reduce the volume of transactions to a more reasonable amount.

Gudahtt commented on app/scripts/lib/setupSentry.js:
```diff
@@ -502,60 +501,87 @@ export default function setupSentry({ release, getState }) {
+     * transactions are sent. By setting `tracesSampleRate` to a value lower than 1.0, we
+     * reduce the volume of transactions to a more reasonable amount.
+     */
+    tracesSampleRate: 0.1,
```
Mobile is using a sampling rate of 0.05, and even that small a rate is a huge amount of events. We should consider adjusting this based upon our current account limits, also taking into consideration the userbase between extension and mobile.

gauthierpetetin replied:
Given `tracesSampleRate` is currently set to 0.04 for Mobile and given that the amount of MAU is more than 3x more important for Extension, I guess we can set `tracesSampleRate` to 0.01 for Extension.

screwball22 reviewed (approved):
JSON v5

gauthierpetetin committed:
fix(sentry): reduce tracesSampleRate value

Given tracesSampleRate is currently set to 0.04 for Mobile and given that the amount of MAU is more than 3x more important for Extension, we can set tracesSampleRate to 0.01 for Extension in order not to exceed our account limits.

gauthierpetetin commented:
Replaced by this PR: https://github.com/MetaMask/metamask-extension/pull/24951
